### PR TITLE
🐛 Live ISO cleanup fix

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
Which issue(s) this PR fixes:
Fixes the cleanup of cluster after live-iso test. Replaced dumpSpecResourcesAndCleanup with just deprovisioning of node, in live iso test we are not creating target cluster, we just provision one node with live ISO image and when test is done we have to deprovision it. Function dumpSpecResourcesAndCleanup was not working because there was no target cluster, so now we are just deprovisioning the live ISO booted node.

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
Will be removed, this is to test CI behaviour. 

/hold

File: test/e2e/live_iso_test.go
Comment (furkatgofurov7) [inline test/e2e/live_iso_test.go:40]: is this needed? this is misleading, we are expecting all bmhs to be in available just before this,

File: test/e2e/live_iso_test.go
Comment (furkatgofurov7) [inline test/e2e/live_iso_test.go:33]:
```suggestion
		By("Waiting for all BMHs to be in Available state")
```

File: test/e2e/live_iso_test.go
Comment (furkatgofurov7) [inline test/e2e/live_iso_test.go:68]:
```suggestion
			g.Expect(isoBmh.Status.Provisioning.State).To(Equal(bmov1alpha1.StateProvisioned), fmt.Sprintf("BMH %s is not in provisioned state", isoBmh.Name))
```

File: test/e2e/live_iso_test.go
Comment (furkatgofurov7) [inline test/e2e/live_iso_test.go:86]:
```suggestion
		By("Deleting live ISO image booted BMH")
```

File: test/e2e/live_iso_test.go
Comment (furkatgofurov7) [inline test/e2e/live_iso_test.go:91]:
```suggestion
		Expect(err).NotTo(HaveOccurred(), "Error getting all BMHs")
```

File: test/e2e/live_iso_test.go
Comment (furkatgofurov7) [inline test/e2e/live_iso_test.go:116]: is it a leftover, I do not find the step where we delete BMH?

File: test/e2e/live_iso_test.go
Comment (furkatgofurov7) [inline test/e2e/live_iso_test.go:98]: it is already set to true once provisioned, you meant false here?

File: test/e2e/live_iso_test.go
Comment (adilGhaffarDev) [reply test/e2e/live_iso_test.go:116]: It should say deprovisioned successfully, I can change it.

File: test/e2e/live_iso_test.go
Comment (adilGhaffarDev) [reply test/e2e/live_iso_test.go:40]: this is just to make sure previous test's deprovisioning is not in process. We thought this might be the issue but now we have checked that the issue was with BMHs Online status, we can remove it, but I think precheck makes the test more robust. Let me know what do you think, should we keep it or remove it?

Comment (furkatgofurov7) [reply test/e2e/live_iso_test.go]: Fine by me to leave it

**Review by furkatgofurov7 — APPROVED** [review]
/approve
/lgtm

Merging it since it has already gotten lgtm (but lost in between) and unblock the e2e tests in CI.

**Pull request merged**
